bills-app/src/main/webapp/script.js

```diff
@@ -112,21 +139,54 @@ function clearBillData() {
  * prints translated text to console
  * param: text to be translated
  */
-function translate(text){
+const translate = async (text, lang, element) => {
     const params = new URLSearchParams();
     params.append('text', text);
-    fetch('/translate', {method: "POST",
-    body: params
-    }).then(response => response.text()).then((translatedText) => {
-        console.log(translatedText);
+    params.append('lang', lang);
+
+    await fetch('/translate', {
+        method: "POST",
+        body: params
+    })
+    .then(response => response.text())
+    .then((translatedText) => {
+        element.innerHTML = translatedText;
   });
 }
 
 /** 
  * calls translate
  * used for body onload
  */
-function addCode(){
-    getRecentBills();
-    translate("This text will be translated"); 
+const addCode = () => {
+    // Setting our default values
+    localStorage.setItem("billType", "recent");
+    localStorage.setItem("lang", "en");
+
+    getRecentBills("en");
+}
+
+const getLanguageOption = (selectObject) => {
+    let selectedLanguage = selectObject.value;
+
+    // Placing the new language value in local storage
+    localStorage.setItem('lang', selectedLanguage);
```

Comment:
I'm not sure how localStorage works, but would it work if in getRecentBills/getUpcomingBills you just read the language from localStorage instead of having it be passed as a param? Or even just read it in translate() because that's where it's really relevant? So here you can just call those functions without passing anything.

Reply:
Ahh yes, you are right. I think I didn't refactor the rest of the code to eliminate any type of redundancy like this one. Thank you for pointing it out!